modules/tinymce/src/core/test/ts/browser/fmt/FormatChangeVarsTest.ts

```diff
@@ -0,0 +1,194 @@
+import { Waiter } from '@ephox/agar';
+import { after, before, beforeEach, context, describe, it } from '@ephox/bedrock-client';
+import { Type } from '@ephox/katamari';
+import { TinyHooks, TinySelections } from '@ephox/mcagar';
+import { assert } from 'chai';
+
+import Editor from 'tinymce/core/api/Editor';
+
+describe('browser.tinymce.core.fmt.FormatChangeVarsTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    base_url: '/project/tinymce/js/tinymce'
+  }, []);
+
+  const events = {
+    general: [] as boolean[],
+    helvetica: [] as boolean[],
+    comicSans: [] as boolean[]
+  };
+
+  let cleanup = null as null | (() => void);
+
+  const clearEvents = () => {
+    events.general = [];
+    events.helvetica = [];
+    events.comicSans = [];
+  };
+
+  const helveticaFont = 'helvetica, arial, sans-serif';
+  const comicSansFont = '\'comic sans ms\', sans-serif';
+
+  before(() => {
+    const editor = hook.editor();
+    const general = editor.formatter.formatChanged('fontname', (evt) => events.general.push(evt), true);
```

Comment:
This would be better suited to using `Singleton.unbindable`.

Reply:
That wasn't what I meant sorry... I meant have 3 variables for each instance, as I want to avoid having the custom cleanup function.